File: staircase/staircase.go
package main

import "fmt"

func staircase(length int32) {
	for stairCount := 1; int32(stairCount) <= length; stairCount++ {
		stair := ""

		spaceCountMax := length - int32(stairCount)
		for spaceCount := 1; spaceCount <= int(spaceCountMax); spaceCount++ {
			stair += " "
		}

		poundCountMax := stairCount
		for poundCount := 1; poundCount <= poundCountMax; poundCount++ {
			stair += "#"
		}

		fmt.Println(stair)
	}
}

File: staircase/Staircase.java
public class Staircase {

  public static void staircase(int length) {
    for (int stairCount = 1; stairCount <= length; stairCount++) {
      String stair = "";

      int spaceCountMax = length - stairCount;
      for (int spaceCount = 1; spaceCount <= spaceCountMax; spaceCount++) {
        stair += " ";
      }

      int poundCountMax = stairCount;
      for (int poundCount = 1; poundCount <= poundCountMax; poundCount++) {
        stair += "#";
      }

      System.out.println(stair);
    }
  }

}
